File: app/src/main/java/com/dumuzeyn/mp3player/TabTransitionRequest.java
package com.dumuzeyn.mp3player;

import android.widget.LinearLayout;
import android.widget.ScrollView;

/** Immutable data required to commit a tab transition. */
final class TabTransitionRequest {
    final int targetTab;
    final int direction;
    final boolean recordHistory;
    final String searchQuery;
    final ScrollView scrollView;
    final LinearLayout content;
    final MainRenderer.PreviewState previewState;

    TabTransitionRequest(int targetTab, int direction, boolean recordHistory,
            String searchQuery, ScrollView scrollView, LinearLayout content,
            MainRenderer.PreviewState previewState) {
        this.targetTab = targetTab;
        this.direction = direction;
        this.recordHistory = recordHistory;
        this.searchQuery = searchQuery == null ? "" : searchQuery;
        this.scrollView = scrollView;
        this.content = content;
        this.previewState = previewState;
    }

    static TabTransitionRequest withoutPreview(int targetTab, int direction,
            boolean recordHistory, String searchQuery) {
        return new TabTransitionRequest(targetTab, direction, recordHistory,
                searchQuery, null, null, null);
    }
}

File: app/src/main/java/com/dumuzeyn/mp3player/TabTransitionRequest.kt
package com.dumuzeyn.mp3player

import android.widget.LinearLayout
import android.widget.ScrollView

/** Immutable data required to commit a tab transition. */
internal class TabTransitionRequest(
    @JvmField val targetTab: Int,
    @JvmField val direction: Int,
    @JvmField val recordHistory: Boolean,
    searchQuery: String?,
    @JvmField val scrollView: ScrollView?,
    @JvmField val content: LinearLayout?,
    @JvmField val previewState: MainRenderer.PreviewState?,
) {
    @JvmField val searchQuery: String = searchQuery.orEmpty()

    companion object {
        @JvmStatic
        fun withoutPreview(
            targetTab: Int,
            direction: Int,
            recordHistory: Boolean,
            searchQuery: String?,
        ): TabTransitionRequest = TabTransitionRequest(
            targetTab,
            direction,
            recordHistory,
            searchQuery,
            null,
            null,
            null,
        )
    }
}
